hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenPauseHandler.java

```diff
@@ -72,12 +72,10 @@ public void handle(@NonNull final TransactionBody txn, @NonNull final WritableTo
         requireNonNull(tokenStore);
 
         var op = txn.tokenPause();
-        var token = tokenStore.get(op.token().tokenNum());
-        if (token.isEmpty()) {
-            throw new HandleException(INVALID_TOKEN_ID);
-        }
+        var token = tokenStore.get(op.tokenOrElse(TokenID.DEFAULT));
+        validateTrue(token != null, INVALID_TOKEN_ID);
```

Comment:
We should probably change `getForModify` also to return `Token`. Can be done in separate PR

Reply:
👍 We can do that along with killing off the `getTokenMeta` method